### PR TITLE
Add shmem_team_ptr

### PR DESCRIPTION
**Pull request by kholland-intel**
Addresses issue #319 and adds a team-based variant to `shmem_ptr`
This PR replaces #459.

### CODE REVIEW

File: content/shmem_team_ptr.tex [hunk 1]
```diff
@@ -0,0 +1,50 @@
+\apisummary{
+    Returns a local pointer to a symmetric data object on the specified \ac{PE} in the specified team.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void *@\FuncDecl{shmem\_team\_ptr}@(shmem\_team\_t team, const void *dest, int pe);
```
[the remark this answers is not in the document]
Comment (kholland-intel) [reply content/shmem_team_ptr.tex:8]: Thank you for catching this @BryantLam.

@jdinan Do you want me to push this change as a doc edit?

[the remark this answers is not in the document]
Comment (jdinan) [reply content/shmem_team_ptr.tex:8]: Yes, please go ahead. Thanks!

[the remark this answers is not in the document]
Comment (jdinan) [reply content/shmem_team_ptr.tex:8]:
```suggestion
void *@\FuncDecl{shmem\_team\_ptr}@(shmem_team_t team, const void *dest, int pe);
```

**Comment by jdinan** [comment]
Merged with the formatting fix -- thank @kholland-intel!

**Pull request merged**